ui/src/components/FormModifications/FormModifications.test.tsx

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { setUnifiedConfig } from '../../util/util';
+import {
+    firstModificationField,
+    getConfigWithModifications,
+    secondModificationField,
+    firstStandardTextField,
+    thirdModificationField,
+    secondStandardTextField,
+} from './TestConfig';
+import EntityModal, { EntityModalProps } from '../EntityModal/EntityModal';
+import { EntitiesWithModifications } from '../BaseFormTypes';
+
+const handleRequestClose = jest.fn();
+const setUpConfigWithDefaultValue = () => {
+    const newConfig = getConfigWithModifications();
+    setUnifiedConfig(newConfig);
+};
+
+const renderModalWithProps = (props: EntityModalProps) => {
+    render(<EntityModal {...props} handleRequestClose={handleRequestClose} />);
+};
+
+const props = {
+    serviceName: 'account',
+    mode: 'create',
+    stanzaName: undefined,
+    formLabel: 'formLabel',
+    page: 'configuration',
+    groupName: '',
+    open: true,
+    handleRequestClose: () => {},
+} satisfies EntityModalProps;
+
+const findMods = (
+    modificationFiled: EntitiesWithModifications,
+    value: string | boolean | number,
+    fieldId: string
+) => {
+    if (modificationFiled?.modifyFieldsOnValue) {
+        return modificationFiled?.modifyFieldsOnValue
+            .find((mod) => mod.fieldValue === value)
+            ?.fieldsToModify.find((field: { fieldId: string }) => field.fieldId === fieldId);
+    }
+    return null;
```

Comment:
is it a valid scenario to have null in mod? Seems like the null checks are scattered across the file

Reply:
removed null but unfortunately undefined is still a valid one